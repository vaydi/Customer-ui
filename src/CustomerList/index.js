/* eslint-disable react/destructuring-assignment */
import { func, number, shape } from 'prop-types';
import {Grid } from '@material-ui/core';
import React from 'react';


import { customerHeadCells, customerTabFiltre } from '../constant/customerConfig';
import TabFilter from '../ConnectedFilter';
import TableDefault from '../Table';
import customersListType from '../types/customersListType';


class CustomersList extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      rowsPerPage: props.rowsPerPage,
      currentPage: props.currentPage,
      token: 'Test_Token',
      tabName: ["ALL"],
    };
  }

  componentDidMount() {
    const { token, currentPage, rowsPerPage } = this.state;
    const { requestCustomerData } = this.props;
    requestCustomerData(token, currentPage, rowsPerPage);
    this.nameCustomerTabFilter(0);
  }

  nameCustomerTabFilter = event => {
    const name = customerTabFiltre.filter(x => x.id === event).map(i => i.name);
    this.setState({ tabName: name });
  };

  changeCurrentPage = page => {
    const { token, rowsPerPage } = this.state;
    const { requestCustomerData } = this.props;
    requestCustomerData(token, page + 1, rowsPerPage);
  };

  changeRowsPerPage = rowsPerPage => {
    const { token } = this.state;
    const { requestCustomerData } = this.props;
    this.setState({ rowsPerPage });
    requestCustomerData(token, 1, rowsPerPage);
  };

  customersDataFormating = customers => {
    const customer = customers.filter(item => {
        if (this.state.tabName[0] !== 'All')
        return item.type === this.state.tabName[0];

        return item;
      })
      .map(x => ({
        amount: `€ ${x.amount}`,
        customerNum: x.customer_num,
        orderRef: x.order_ref,
        creationDate: x.creation_date,
        paymentOption: x.payment_option.name,
        type: x.type,
        status: x.status,
      }));
    return customer;
  };

  render() {
    const { customers = [], pagination } = this.props.listeCustomer;
    const customersDataFormating = this.customersDataFormating(customers);

    return (
      <Grid item xs={12}>
        <TabFilter
          TabType={customerTabFiltre}
          onClickTab={event => this.nameCustomerTabFilter(event)}
        />
        {
          <TableDefault
            headCells={customerHeadCells}
            data={customersDataFormating}
            // extraDataFormating={customers}
            noDataMessage="Aucune customer trouvée"
            changePage={event => this.changeCurrentPage(event)}
            changeRowsPerPage={event => this.changeRowsPerPage(event)}
            pagination={pagination}
          />
        }
      </Grid>
    );
  }
}

CustomersList.defaultProps = {
  listeCustomer: {
    customers: [],
    pagination: {},
  },
  requestCustomerData: Function.prototype,
  currentPage: 1,
  rowsPerPage: 20,
};

CustomersList.propTypes = {
  listeCustomer: shape(customersListType),
  requestCustomerData: func,
  currentPage: number,
  rowsPerPage: number,
};

export default CustomersList;
