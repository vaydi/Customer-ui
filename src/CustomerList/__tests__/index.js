import itRendersAllMutations from 'jest-it-renders-all-mutations';

import CustomerList from '..';

const mutations = [
  {
    name: 'with data',
    props: {
      listeCustomer: {
        customers: [
          {
            amout: 10,
            creation_date: '2019-11-07T10:47:35.526Z',
            order_ref: 'CP004822587',
            payment_option: {
              id: 154,
              name: 'Carte bancaire 3D 4X (Eureka)',
            },
            status: 'Undetermined',
            customer_num: 1,
            type: 'Autorisation',
          }
        ],
        pagination: { current_page: 1, nb_result: 100, result_by_page: 3, total_page: 33 },
      },
      requestCustomerData: jest.fn(),
      match: { params: {}, isExact: true, path: '', url: '' },
      location: {pathname: "/", search: "", hash: "", state: undefined}
    },
  },
  {
    name: 'without data',
    props: {
      listeCustomer: { customers: [], pagination: {} },
      requestCustomerData: jest.fn(),
    },
  },
];

describe(CustomerList.name, () => {
  itRendersAllMutations(CustomerList, mutations);
});
