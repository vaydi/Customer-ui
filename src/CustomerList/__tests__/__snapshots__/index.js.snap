// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomersList defaultProp functions runs defaultProps.requestCustomerData() 1`] = `undefined`;

exports[`CustomersList with data renders correctly 1`] = `
<div>
  <h1>
    List Customers 
  </h1>
</div>
`;

exports[`CustomersList without data renders correctly 1`] = `
<div>
  <h1>
    List Customers 
  </h1>
</div>
`;
